### ISSUE TITLE
Add support for direct use of geographic data

### ISSUE THREAD

**Issue by alex-s-gardner** [Deltares/Geomorphometry.jl#33]
I'd like to calculate slope and curvature from a geographic DEM (raster dimensions are in degrees). I'd like to do this without retrojecting the DEM. I think the only way to do this is to calculate component values (slope/curvature along lat/lon) then to scale each component by the local meters per degree lat/lon. Given this, would it make sense to add an option to relevant functions to return component values?

**Comment by evetion** [comment]
Yeah, this is something we need. A weighting factor for the y/x could be provided, probably as a matrix of the same size as the dem?

**Comment by alex-s-gardner** [comment]
Ya, a row to column scaling would work. This could be a scalar (rectangular cells) or matrix the same size as the DEM (geographic).

**Comment by alex-s-gardner** [comment]
should we introduce slope components (x and y) as a new function (maybe slopexy) of do we add a kwarg to slope?

do we adopt the convention that slope across columns to be x_slope and slope across rows to be y_slope?

**Comment by evetion** [comment]
Hmm. I was thinking of passing a x/y ratio as an argument to most methods. It can be a single value, or a column, or a matrix. The value and columns should be converted to a matrix so it can be used in the calculation.

Ideally, we can automatically set these things, if we have a trait for geographic stuff in GeoInterface. Let me whip up a PoC.

**Comment by alex-s-gardner** [comment]
That sounds good to me and would be the use case for most users. For my workflows I will still need slope in x and slope in y, and curvature in x and curvature in y. I could back out slope x/y from slope and azimuth but I suspect it's faster just to compute them directly. But i think it will be strait forward to add those after the geographic stuff is implemented. 

The idea of automating this sounds like a dream.

**Comment by alex-s-gardner** [comment]
Will do... we could add a "dims" kwarg to slope

**Comment by evetion** [comment]
Fixed in #41, we introduce a cellsize tuple (default (1,1)) argument that's automatically set for GeoArrays and Rasters. See an example here: https://github.com/Deltares/Geomorphometry.jl/blob/main/ext/GeomorphometryRastersExt.jl (note the scaling by 111_000 for geographic rasters).

This is a substantial improvement, but given the cellsize is a fixed argument, rasters spanning multiple degrees will become more incorrect at the edges.
